### PR TITLE
Make Link join Lobby

### PR DESCRIPTION
**Pull request by wtachau**
[[M-L] Improve Host/Join Experience](https://trello.com/c/NwI9t8lG/6-m-l-improve-host-join-experience)

### CODE REVIEW

File: app/src/components/GameRoutes.tsx [hunk 1]
```diff
@@ -15,14 +25,54 @@ function CurrentPlayerProvider(props: {
...
 }) {
-  const { data, loading } = useCurrentPlayerQuery({
+  const [joinGame] = useJoinGameMutation()
+  const [getPlayersResponse] = useCurrentPlayerLazyQuery({
     variables: {
       joinCode: props.joinCode,
       clientUuid: clientUuid()
+    },
+    onCompleted: async data => {
+      console.log("got players response data:", data)
```
Comment (wtachau) [inline app/src/components/GameRoutes.tsx:35]: this never gets called -- the `joinGame` query causes an infinite render first

File: app/src/components/GameRoutes.tsx [hunk 2]
```diff
@@ -15,14 +25,54 @@ function CurrentPlayerProvider(props: {
...
 
-  if (!loading && !data?.players[0]) {
+  const playersResponse = useCurrentPlayerQuery({
+    variables: {
+      joinCode: props.joinCode,
+      clientUuid: clientUuid()
+    }
+  })
+
+  const gamesResponse = useGameByJoinCodeQuery({
```
Comment (wtachau) [inline app/src/components/GameRoutes.tsx:46]: i imagine we only want to make this call if we don't see games in the `playersResponse`, but then I get an error about React hooks getting called conditionally

**Comment by avimoondra** [comment]
[[M-L] Improve Host/Join Experience](https://trello.com/c/NwI9t8lG/6-m-l-improve-host-join-experience)

File: graphql-server/migrations/1588378316564_update_permission_anonymous_public_table_players/up.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- args:
+    permission:
+      allow_aggregations: false
+      columns:
+      - client_uuid
+      - game_id
+      - id
+      - team
+      - team_sequence
+      - username
```
Comment (wtachau) [inline graphql-server/migrations/1588378316564_update_permission_anonymous_public_table_players/up.yaml:16]: @avimoondra I was getting an error when fetching the current player, since it looks like the anonymous user doesn't have access to the `username` or `team_sequence` field.

That error happens currently if you try to go to a game lobby when don't have a user stored in localStorage:
Play_the_Fishbowl_Game__aka_Salad_Bowl__-_Online___Free

It looks like, when we were using the non-lazy version of this query (`const { data, loading } = useCurrentPlayerQuery({`),  the way to error handle was just to also unpack the `error` object (`const { data, loading, error } = ...`).

With the lazy version, you need to pass an `onError` callback, which means when there _is_ an error, the [`onComplete` callback](https://github.com/avimoondra/fishbowl/pull/22/files#diff-df1ab9a06b4534c03fc258438684a7afR52) doesn't fire. 

So tl;dr I'm pretty sure we need to fix these permissions issues so that we can properly handle fetching the current user when there is none.

File: app/src/components/GameRoutes.tsx [hunk 4]
```diff
@@ -15,23 +27,88 @@ function CurrentPlayerProvider(props: {
...
+      }
+
+      setSkipCallback(true)
+      setAttemptedGameLoad(true)
     }
   })
 
-  if (!loading && !data?.players[0]) {
+  // To start, fetch the current player and the games they are a part of
+  React.useEffect(() => {
```
Comment (wtachau) [inline app/src/components/GameRoutes.tsx:82]: this replaces the initial ` const { data, loading } = useCurrentPlayerQuery({` query

**Comment by avimoondra** [comment]
closing in favor of #24
